Automattic/wp-calypso
# Plugins: Reset category selection when searching

Fixes #66938

## Proposed Changes

When searching on the plugins browser input:
* Assure it's on the main Plugins page before constructing the URL with the search param. If not, first redirect to the main Plugins page.
* When on a category page, reset breadcrumbs to allow them to be reconstructed only with the search

## Testing Instructions

* On this branch
* Go to the `/plugins` page
* Select a category
* Search any term on the input
* The URL should NOT contain the category term
* Click on the `X` button on the input
* It should redirect to the main Plugins page
* On production the behavior should be the same as described here: #66938

| Before  | After |
| ------------- | ------------- |
|![CleanShot 2023-07-03 at 15 38 06](https://github.com/Automattic/wp-calypso/assets/5039531/de074a4a-d464-4cbd-9270-8a5cea917e33)|![CleanShot 2023-07-03 at 15 49 55](https://github.com/Automattic/wp-calypso/assets/5039531/b82c7d95-767f-4ec7-a9ca-398f3c95084f)|


## Pre-merge Checklist
